synapse/media/media_repository.py

```diff
@@ -373,15 +379,17 @@ async def _get_remote_media_impl(
 
         # If we have an entry in the DB, try and look for it
         if media_info:
-            file_id = media_info["filesystem_id"]
+            file_id = media_info.filesystem_id
             file_info = FileInfo(server_name, file_id)
 
-            if media_info["quarantined_by"]:
+            if media_info.quarantined_by:
                 logger.info("Media is quarantined")
                 raise NotFoundError()
 
-            if not media_info["media_type"]:
-                media_info["media_type"] = "application/octet-stream"
+            if not media_info.media_type:
+                media_info = attr.evolve(
+                    media_info, media_type="application/octet-stream"
+                )
```

Comment:
Is this to avoid mutating the cache contents? If so, is it worth making `get_cached_remote_media` fill in this default value for us?

Reply:
The value isn't cached (the "cache" in `get_cached_remote_media` means "downloaded by this server", not "in an LRU cache").

That seems inappropriate for a database method, but might make sense if we *always* do that?

Reply:
> That seems inappropriate for a database method, but might make sense if we _always_ do that?

That was my thinking---would make it easier to accidentally forget filling in a default media type. Probably best to just leave this as-is though.

Reply:
The other option would be to not make the attrs frozen, but that feels kind of meh as we *usually* do that? What do you think?

Reply:
Ahhhh I'd missed that the attrs was frozen. (Was wondering why we needed to `.evolve()`.

I'm very much in favour of keeping things frozen!

Reply:
OK, going to keep this as is then.